Documentation/Evolution/CharacterClasses.md

```diff
@@ -0,0 +1,452 @@
+  ///     // Prints:
+  ///     // 1 --> Optional(1)
+  ///     // ९ --> Optional(9)
+  ///     // A --> nil
+  public var decimalDigitValue: Int? { get }
+
+}
+
+extension Unicode.Scalar {
+  /// A Boolean value indicating whether this scalar is considered 
+  /// a decimal digit.
+  ///
+  /// Any Unicode scalar that has a `numericType` property equal to `.decimal`
+  /// is considered a decimal digit. This includes the digits from the ASCII
+  /// range, from the _Halfwidth and Fullwidth Forms_  Unicode block, as well
+  ///  as digits in some scripts, like `DEVANAGARI DIGIT NINE` (U+096F).
+  public var isDecimalDigit: Bool { get }
+
+  // TODO: Should we provide the value too? what about other Character properties?
+}
+```
+
+`\D` matches the inverse of `\d`.
+
+_<details><summary>Rationale</summary>_
+
+Unicode's recommended definition for `\d` is its [numeric type][numerictype] of "Decimal" in contrast to "Digit". It is specifically restricted to sets of ascending contiguously-encoded scalars in a decimal radix positional numeral system. Thus, it excludes "digits" such as superscript numerals from its [definition][derivednumeric] and is a proper subset of `Character.isWholeNumber`. 
+
+We interpret Unicode's definition of the set of scalars, especially it's requirement that scalars be encoded in ascending chains, to imply that this class is restricted to scalars which meaningfully encode base-10 digits. Thus, we choose to make this Character property _restrictive_, similar to `isHexDigit` and `isWholeNumber` and provide a way to access this value.
+
+It's possible we might add future properties to differentiate Unicode's non-decimal digits, but that is outside the scope of this pitch.
+
+</details>
+
+### Word characters: `\w`, `\W`
+
+We propose `\w` be named "word character" with the following definitions:
+
+```swift
+extension Character {
+  /// A Boolean value indicating whether this scalar is considered 
+  /// a "word" character.
+  ///
+  /// All characters with an initial Unicode scalar that is considered a
+  /// word character are considered word character.
+  public var isWordCharacter: Bool { get }    
+}
+
+extension Unicode.Scalar {
+  /// A Boolean value indicating whether this scalar is considered 
+  /// a "word" character.
+  ///
+  /// Any Unicode scalar that has one of the Unicode properties
+  /// `Alphabetic`, `Digit`, or `Join_Control`, or is in the 
+  /// general category `Mark` or `Connector_Punctuation`.
+  public var isWordCharacter: Bool { get }
+}
+```
+
+`\W` matches the inverse of `\w`.
+
+_<details><summary>Rationale</summary>_
+
+The Unicode recommendation is to derive word character matching from Unicode properties and general categories as described above
+
+We chose to treat any grapheme cluster that leads with a Unicode scalar word character as a word character as well.
+
+**TODO** How does this relate to `isLetter`? Why don't we have `alphaNumeric`? Let's establish proper subsetting here.
```

Comment:
`isLetter` means any scalar that is part of the Letter parent category (Lu, Ll, Lt, Lm, Lo). That's a subset of `Alphabetic`, which also includes the Nl / Letter Numbers category and `Other_Alphabetic`, which is defined in PropList.txt.

I don't see `alphaNumeric` defined as a Unicode property, but it looks like it generally is what we've defined as a word character without the punctuation, marks, and connectors.